File: InformationFinder/src/main/java/org/aksw/provenance/ml/feature/fact/impl/NameFeature.java
/**
 * 
 */
package org.aksw.provenance.ml.feature.fact.impl;

import java.util.Set;

import org.aksw.provenance.evidence.ComplexProof;
import org.aksw.provenance.evidence.Evidence;
import org.aksw.provenance.ml.feature.fact.AbstractFactFeatures;
import org.aksw.provenance.ml.feature.fact.FactFeature;


/**
 * @author Daniel Gerber <dgerber@informatik.uni-leipzig.de>
 *
 */
public class NameFeature implements FactFeature {

    /* (non-Javadoc)
     * @see org.aksw.provenance.ml.feature.fact.FactFeature#extractFeature(org.aksw.provenance.evidence.ComplexProof, java.util.Set)
     */
    @Override
    public void extractFeature(ComplexProof proof, Evidence evidence) {

        proof.getFeatures().setValue(AbstractFactFeatures.SUBJECT, proof.getSubject());
        proof.getFeatures().setValue(AbstractFactFeatures.PHRASE, proof.getProofPhrase().replaceAll("\\n", ""));
        proof.getFeatures().setValue(AbstractFactFeatures.OBJECT, proof.getObject());
        proof.getFeatures().setValue(AbstractFactFeatures.CONTEXT, proof.getContext().replaceAll("\\n", ""));
    }
}

File: python/old/triple/fact_extraction/NameFeature.py
# -*- coding: utf-8 -*-
"""
@author: JohannesLoevenich
"""
from AbstractFactFeature import AbstractFactFeature


class NameFeature(AbstractFactFeature): 
     def set_values(self,subject, phrase, object,context,file_name, language):
        self.subject = subject
        self.phrase = phrase
        self.object = object
        self.context = context
        self.file_name = file_name
        self.language = language
 
        
